Lesson 4: Add appeal form (web & api)

Skurik-SA opened the pull request.

Skurik-SA committed:
Lesson 3: Add news (api & web) and pages (api)

Сделан Lesson 3

Skurik-SA committed:
Lesson 4

Skurik-SA committed:
Lesson4 HM

chrn-feip commented:
@Skurik-SA сделайте ребейз ветки, чтобы остались коммиты только по ДЗ 4. И конфликты пофиксите.

chrn-feip commented on fefu-app2/app/Http/Controllers/Web/AppealWebController.php:
```diff
@@ -0,0 +1,31 @@
+
+        $appeal = new Appeal();
+        $appeal->name = $data['name'];
+        $appeal->phone = PhoneSanitizer::sanitize($data['phone']??null);
```
Здесь нет необходиомсти писать ?? null, т.к. в веб-контроллере все поля будут в запросе, потому что они все есть в форме на странице. В апи мы это пишем, потому что запросто можем телефон и/или почту не указывать в параметрах запроса.

chrn-feip replied:
Почты строкой ниже это тоже касается

chrn-feip replied:
@Skurik-SA

Skurik-SA replied:
Убрал

chrn-feip commented on fefu-app2/app/Http/Requests/StoreAppealRequest.php:
```diff
@@ -0,0 +1,30 @@
+
+use Illuminate\Foundation\Http\FormRequest;
+
+class StoreAppealRequest extends FormRequest
```
Это не нужно

chrn-feip commented on fefu-app2/app/Http/Requests/UpdateAppealRequest.php:
```diff
@@ -0,0 +1,30 @@
+
+use Illuminate\Foundation\Http\FormRequest;
+
+class UpdateAppealRequest extends FormRequest
```
Это не нужно

chrn-feip commented on fefu-app2/app/Policies/AppealPolicy.php:
```diff
@@ -0,0 +1,94 @@
+use App\Models\User;
+use Illuminate\Auth\Access\HandlesAuthorization;
+
+class AppealPolicy
```
У нас нет никаких ограничений по тому, кто может оставлять обратную связь. Соответственно, и нет  необходимости вводить класс политики, который всем всё разрешает.

chrn-feip commented on fefu-app2/database/factories/AppealFactory.php:
```diff
@@ -0,0 +1,23 @@
+/**
+ * @extends \Illuminate\Database\Eloquent\Factories\Factory<\App\Models\Appeal>
+ */
+class AppealFactory extends Factory
```
Не нужно

chrn-feip commented on fefu-app2/database/seeders/AppealSeeder.php:
```diff
@@ -0,0 +1,19 @@
+use Illuminate\Database\Console\Seeds\WithoutModelEvents;
+use Illuminate\Database\Seeder;
+
+class AppealSeeder extends Seeder
```
Не нужно

Skurik-SA committed:
Правки

Ну, собственно, все эти файлы были созданы автоматически, делал всё прям как на паре, вот они и вылезли тут.
Что было подмечено как ненужное -- всё было вырезано

Skurik-SA committed:
Убрал дублирование

Skurik-SA committed:
Revert "Lesson4 HM"

This reverts commit e87dd7baf7c90688ae7a8858ab122b93f8f12611.

Skurik-SA committed:
Я всё сломал там, но теперь всё починил, правки были сделаны...

Skurik-SA committed:
Убрал у почты
